### PR TITLE
Add maxExecutors configuration for streaming queries

### PR DESCRIPTION
**Pull request by penghuo**
### Description
* add a new conf `spark.flint.streaming.dynamicAllocation.maxExecutors` (default 10) to control streaming query max executors count. it will override spark.dynamicAllocation.maxExecutors.
* it is short-term solution, in long-term client application should configure `spark.dynamicAllocation.maxExecutors` directly. Track with https://github.com/opensearch-project/sql/issues/2641
* test with EMR-S
  * spark.flint.streaming.dynamicAllocation.maxExecutors = 5, max 5 executors allocated for streaming query.
  * no setting, max 10 executors allocated for streaming query.

### Issues Resolved
https://github.com/opensearch-project/opensearch-spark/issues/324

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spark-sql-application/src/main/scala/org/apache/spark/sql/FlintJobExecutor.scala [hunk 1]
```diff
@@ -85,6 +85,19 @@ trait FlintJobExecutor {
         "org.opensearch.flint.spark.FlintPPLSparkExtensions,org.opensearch.flint.spark.FlintSparkExtensions")
   }
 
+  /*
+   * Override dynamicAllocation.maxExecutors with streaming maxExecutors. more detail at
+   * https://github.com/opensearch-project/opensearch-spark/issues/324
+   */
+  def configDYNMaxExecutors(conf: SparkConf, jobType: String): Unit = {
```
Comment (noCharger) [inline spark-sql-application/src/main/scala/org/apache/spark/sql/FlintJobExecutor.scala:92]: nit: configDYNMaxExecutors -> configDRAMaxExecutors?

**Review by noCharger — APPROVED** [review]

File: spark-sql-application/src/main/scala/org/apache/spark/sql/FlintJobExecutor.scala (hunk 1 above)
Comment (penghuo) [reply spark-sql-application/src/main/scala/org/apache/spark/sql/FlintJobExecutor.scala:92]: DYN is prefix of dynamic allocation configurations.

```
  private[spark] val DYN_ALLOCATION_MAX_EXECUTORS =
    ConfigBuilder("spark.dynamicAllocation.maxExecutors")
      .version("1.2.0")
      .intConf
      .createWithDefault(Int.MaxValue)
```

**Review by seankao-az — APPROVED** [review]

File: integ-test/src/test/scala/org/apache/spark/sql/FlintJobITSuite.scala [hunk 2]
```diff
@@ -218,6 +217,43 @@ class FlintJobITSuite extends FlintSparkSuite with JobTest {
...
+    val query =
+      s"""
+         | CREATE SKIPPING INDEX ON $testTable
+         | (
+         |   year PARTITION,
+         |   name VALUE_SET,
+         |   age MIN_MAX
+         | )
+         | WITH (auto_refresh = true)
+         | """.stripMargin
```
Comment (seankao-az) [inline integ-test/src/test/scala/org/apache/spark/sql/FlintJobITSuite.scala:230]: not sure which streaming jobs this config need for?
I think creating auto refresh index will not go thru the FlintJob / FlintREPL. Do you mean to configDYNMaxExecutors in AutoIndexRefresh?

Comment (penghuo) [reply integ-test/src/test/scala/org/apache/spark/sql/FlintJobITSuite.scala]: remove IT , add UT.

the configuration supposed to work for all streaming query.

Comment (seankao-az) [reply integ-test/src/test/scala/org/apache/spark/sql/FlintJobITSuite.scala]: > supposed to work for all streaming query

Including these auto refresh streaming job?
Previously these two test cases failed: https://github.com/opensearch-project/opensearch-spark/actions/runs/8902308394/job/24447980601
Screenshot 2024-04-30 at 4 57 22 PM

**Review by seankao-az — APPROVED** [review]

**Pull request merged**